s3storagebasedauthorizationmanager/src/main/java/org/sfdc/uip/hive/ql/security/authorization/URMCredentialsRetriever.java

```diff
@@ -0,0 +1,71 @@
+package org.sfdc.uip.hive.ql.security.authorization;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.BasicSessionCredentials;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.net.URLConnection;
+
+public class URMCredentialsRetriever
+{
+    private static final String URM_ADDRESS_FOR_IMPERSONATION = "http://localhost:9944/latest/meta-data/iam/security-credentials/impersonation/";
+    private static final Log LOG = LogFactory.getLog(URMCredentialsRetriever.class);
+
+    AWSCredentials getCredentialsForUser(String userName) {
+        try {
+            //Create http-client to get user mapped role credentials
+            URI uri = URI.create(URM_ADDRESS_FOR_IMPERSONATION + userName);
+            URLConnection connection = uri.toURL().openConnection();
+            InputStream response = connection.getInputStream();
+
+            try (BufferedReader rd = new BufferedReader(new InputStreamReader(response))) {
+                StringBuilder responseString = new StringBuilder(); // or StringBuffer if Java version 5+
+                String line;
+                while ((line = rd.readLine()) != null) {
+                    responseString.append(line);
+                    responseString.append('\r');
+                }
```

Comment:
We can probably simplify this to:
String line = rd.lines().collect(Collectors.joining("\n"))

Reply:
Thanks for checking! It makes sense.